### ISSUE TITLE
Configure prefers my Visual Studio 2008 Express tools instead

### ISSUE THREAD

**Issue by Ian502** [FWGS/xash3d-fwgs#851]
Whenever I try to configure the project it uses the compiler/dependencies from MSVS 2008 Express instead of the newer MSVS 2017 Community.

Even when I checked the log, and it runs "vswhere.exe" from the  2017's folder:

**The Config.Log output:**
`Checking for 'msvc' (C compiler)
['C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vswhere.exe', '-products', '*', '-legacy', '-format', 'json']
out: [

  {

    "instanceId": "fe0ff76e",

    "installDate": "2022-05-13T03:00:31Z",

    "installationName": "VisualStudio/15.9.48+28307.1974",

    "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community",

    "installationVersion": "15.9.28307.1974",

    "productId": "Microsoft.VisualStudio.Product.Community",

    "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\Common7\\IDE\\devenv.exe",`

**What CMD shows:**

```G:\Dev\Xash3d\xash3d-fwgs>waf configure -T release --sdl2=g:/xash3d/xash3d-fwgs/sdl2
Página de códigos activa: 1252
Using python
Setting top to                                          : G:\Dev\Xash3d\xash3d-fwgs
Setting out to                                          : G:\Dev\Xash3d\xash3d-fwgs\build
Build type                                              : release
LTO build                                               : no
PolyOpt build                                           : no
Checking for 'msvc' (C compiler)                        : C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\CL.exe
Target OS                                               : win32
Target CPU                                              : x86
Target binfmt                                           : pe
Checking for 'msvc' (C++ compiler)                      : C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\CL.exe
Target OS                                               : win32
Target CPU                                              : x86
Target binfmt                                           : pe
Checking for program 'git'                              : C:\Program Files\Git\cmd\git.exe
Checking git hash                                       : 81b01ac
Checking if 'msvc' can target 32-bit                    : yes
Checking for header stdint.h                            : not found
Checking for MSVC library                               : user32
Checking for MSVC library                               : shell32
Checking for MSVC library                               : gdi32
Checking for MSVC library                               : advapi32
Checking for MSVC library                               : dbghelp
Checking for MSVC library                               : psapi
Checking for MSVC library                               : ws2_32
Checking for header tgmath.h                            : not found
Checking for header alloca.h                            : not found
Checking for header malloc.h                            : not found
--> public                                              : in progress
<-- public                                              : done
--> game_launch                                         : in progress
<-- game_launch                                         : done
--> ref_gl                                              : in progress
<-- ref_gl                                              : done
--> ref_soft                                            : in progress
<-- ref_soft                                            : done
--> mainui                                              : in progress
Checking if 'msvc' supports C++11                       : no
<-- mainui                                              : done
--> vgui_support                                        : in progress
Does this architecture support VGUI?                    : yes
Does this OS support VGUI?                              : yes
Does this toolchain able to link VGUI?                  : yes
Configuring VGUI by provided path                       : yes: ['vgui'], ['G:\\Dev\\Xash3d\\xash3d-fwgs\\vgui-dev\\lib\\win32_vc6'], ['G:\\Dev\\Xash3d\\xash3d-fwgs\\vgui-dev\\include']
<-- vgui_support                                        : done
--> engine                                              : in progress
Configuring SDL2 by provided path                       : yes: ['SDL2'], ['g:\\xash3d\\xash3d-fwgs\\sdl2\\lib\\x86'], ['g:\\xash3d\\xash3d-fwgs\\sdl2\\include', 'g:\\xash3d\\xash3d-fwgs\\sdl2\\include\\SDL2']
Checking for library SDL2 sanity                        : no
SDL2 not availiable! If you want to build dedicated server, specify --dedicated
(complete log in `G:\Dev\Xash3d\xash3d-fwgs\build\config.log)
```

I've also tried multiple times telling the waf configure, which version of MSVS I'm using. If I specify 15.0 or 16.0, it chooses the 9.0 (2008) if I pick other versions, it defaults to mingw (which doesn't configure correctly either)

I'm trying to compile for x86, as my mod relies heavily on VGUI, i want to make it XASH Compatible, and build it as a standalone mod. Any help on what may be causing this?

Thanks!

**Comment by Ian502** [comment]
I want to build Xash3d so that I can ship it as a standalone game.

I have done what you said to no avail. However I have detected that when running with the -8 flag, it checks for my community directory but, I need it to be 32bit not 64bit, because I use VGUI.

![imagen](https://user-images.githubusercontent.com/11865784/168340535-120c545e-ed67-4a31-9f92-8273f254e323.png)

But it still uses the VS 2008 Express CL.EXE for X86

![imagen](https://user-images.githubusercontent.com/11865784/168340674-72aee222-4c6d-4afc-887b-3f02fab13271.png)


And I'm skipping sanity check on SDL, because for some reason it tells me it wasn't found :c 

Thanks!

**Comment by a1batross** [comment]
>I want to build Xash3d so that I can ship it as a standalone game.

Then you only need to modify game_launch and build it. It's simple and just one file without any dependencies. :)

>But it still uses the VS 2008 Express CL.EXE for X86

You can also try pass `--msvc_version="msvc 15.9"` in configuration stage. This will force waf to pick selected compiler.

**Comment by a1batross** [comment]
Fixed in upstream waf, fixed in fd83b0dc17e4262c7ea7176b3db42a52c48e0b56
